coordinates/Coordinates/DirectionCoordinate.cc

```diff
@@ -1619,88 +1619,81 @@ Bool DirectionCoordinate::toMix2(Vector<Double>& out,
 //
     return True;
 }
-     
-
-
-
-Coordinate* DirectionCoordinate::makeFourierCoordinate (const Vector<Bool>& axes,
-                                                        const Vector<Int>& shape) const
-//
-// axes says which axes in the coordinate are to be transformed
-// shape is the shape of the image for all axes in this coordinate
-//
-{
-   AlwaysAssert(nPixelAxes()==2, AipsError);
-   AlwaysAssert(nWorldAxes()==2, AipsError);
-//
-   if (axes.nelements() != 2) {
-      set_error ("Invalid number of specified axes");
-      return 0;
-   }
-   if (!axes[0] || !axes[1]) {
-      set_error ("You must specify both axes of the DirectionCoordinate to transform");
-      return 0;
-   }
-//
-   if (shape.nelements() != 2) {
-      set_error ("Invalid number of elements in shape");
-      return 0;
-   }
 
-// Find names and units for Fourier coordinate and units to set 
-// for this DirectionCoordinate 
-
-   Vector<String> names(worldAxisNames());
-   Vector<String> units(worldAxisUnits());
-   Vector<String> unitsCanon(worldAxisUnits());
-//
-   Vector<String> namesOut(worldAxisNames().copy());
-   Vector<String> unitsOut(worldAxisUnits().copy());
-   fourierUnits (namesOut[0], unitsOut[0], unitsCanon[0], Coordinate::DIRECTION, 0,
-                 units[0], names[0]);
-   fourierUnits (namesOut[1], unitsOut[1], unitsCanon[1], Coordinate::DIRECTION, 1,
-                 units[1], names[1]);
-
-// Make a copy of ourselves and set the new units
+Coordinate* DirectionCoordinate::makeFourierCoordinate(
+    const Vector<Bool>& axes, const Vector<Int>& shape
+) const {
+    // axes says which axes in the coordinate are to be transformed
+    // shape is the shape of the image for all axes in this coordinate
+    AlwaysAssert(nPixelAxes()==2, AipsError);
+    AlwaysAssert(nWorldAxes()==2, AipsError);
+
+    if (axes.nelements() != 2) {
+        set_error ("Invalid number of specified axes");
+        return 0;
+    }
+    if (!axes[0] || !axes[1]) {
+        set_error ("You must specify both axes of the DirectionCoordinate to transform");
+        return 0;
+    }
+    if (shape.nelements() != 2) {
+        set_error ("Invalid number of elements in shape");
+        return 0;
+    }
 
-   DirectionCoordinate dc = *this;
-   if (!dc.setWorldAxisUnits(unitsCanon)) {
-      set_error ("Could not set world axis units");
-      return 0;
-   }
+    // Find names and units for Fourier coordinate and units to set 
+    // for this DirectionCoordinate 
+    Vector<String> names = worldAxisNames().copy();
```

Comment:
I agree and did this originally. However, Vector<String>.copy() (the RHS) returns an Array<String>, because copy() is implemented in Array<> and Vector does not override it, so the type must be explicitly declared in this case so the implicit cast happens, unless I'm missing something.